src/bloqade/stim/rewrite/ifs_to_stim.py

```diff
@@ -139,8 +137,13 @@ def rewrite_Statement(self, node: ir.Statement) -> RewriteResult:
 
     def rewrite_IfElse(self, stmt: scf.IfElse) -> RewriteResult:
 
-        # Check the condition is a singular MeasurementIdBool
-        if not isinstance(self.measure_frame.entries[stmt.cond], MeasureIdBool):
+        condition_type = self.measure_frame.entries[stmt.cond]
```

Comment:
This will raise a `KeyError` if `stmt.cond` isn't found in `entries`. I think you should use `self.measure_frame.get(stmt.cond)`. The `isinstance(condition_type, MeasureIdBool)` check below will handle the `None` case.